### PR TITLE
[processing] Interpolation chapter

### PR DESCRIPTION
**Pull request by ghtmtt**
This PR implements all the 3 interpolations tool in Processing. 

The Heatmap example has been reviewed in order to be consistent with the Processing dialogs, but the sample data and the resulting pictures are the same (pictures have been moved to the correct `resource` folder of processing_algorithm)

When this PR is merged we can also merge #1806

### CODE REVIEW

File: source/docs/user_manual/processing_algs/qgis/interpolation.rst [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+``Number of columns`` [number]
+  Total number of columns of the output raster layer.
+
+``Number of rows`` [number]
+  Total number of columns of the output raster layer.
+
+``Extent (xmin, xmax, ymin, ymax)`` [extent]
+  Extent of the output raster layer. You have to declare the output extent by
+  either choosing it from the map canvas of selecting it from another layer.
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/interpolation.rst]: either using a layer or the map canvas extent or selecting it from the map canvas (to use same options than in the dialog - and imho it gives more options than described)

File: source/docs/user_manual/processing_algs/qgis/interpolation.rst [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+as possible triangles. The resulting surfaces are not smooth.
+
+The algorithm creates both the raster layer of the interpolated values and the
+vector line layer with the triangulation boundaries.
+
+Parameters
+..........
+
+``Vector layer`` [vector: point]
+  Point vector layer to use for the heatmap
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/interpolation.rst]: heatmap?

File: source/docs/user_manual/processing_algs/qgis/interpolation.rst [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+
+``Interpolation attribute`` [tablefield: numeric]
+  Field used for the interpolation. Once you have chosen the layer and the field
+  click on the |signPlus| button to add the layer and the field to the list.
+  Click on the |signMinus| button to remove the layer from the list.
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/interpolation.rst]: same as IDW

File: source/docs/user_manual/processing_algs/qgis/interpolation.rst [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+
+``Use-Z-coordinate for interpolation`` [checkbox]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/interpolation.rst]: use Z-coordinate

File: source/docs/user_manual/processing_algs/qgis/interpolation.rst [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+  * Linear
+  * Clough-Toucher (cubic)
+
+  Default: *Linear*
+
+``Number of columns`` [number]
+  Total number of columns of the output raster layer.
+
+``Number of rows`` [number]
+  Total number of columns of the output raster layer.
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/interpolation.rst]: rows

File: source/docs/user_manual/processing_algs/qgis/interpolation.rst [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+
+``Extent (xmin, xmax, ymin, ymax)`` [extent]
+  Extent of the output raster layer. You have to declare the output extent by
+  either choosing it from the map canvas of selecting it from another layer.
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/interpolation.rst]: same as IDW

File: source/docs/user_manual/processing_algs/qgis/index.rst [hunk 7]
```diff
@@ -20,6 +20,7 @@ algorithms.
      database
      file_tools
      layer_tools
+     interpolation
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/index.rst]: is it an alphabetic classement ?--> move up  this section once

File: source/docs/user_manual/processing_algs/qgis/interpolation.rst [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+
+  Default: *Raw*
+
+Output
+......
+``Heatmap`` [raster]
+  Raster layer with kernel density values
+
+
+Example: Creating a Heatmap
```
[the remark this answers is not in the document]
Comment (ghtmtt) [reply source/docs/user_manual/processing_algs/qgis/interpolation.rst]: agree, but, if you don't mind, I'll leave it here for the moment. we can move it in the future

File: source/docs/user_manual/processing_algs/qgis/interpolation.rst [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+  increase the influence certain features have on the resultant heatmap.
+
+``Kernel shape`` [selection]
+  Controls the rate at which the influence of a point decreases as the distance
+  from the point increases. Different kernels decay at different rates, so a triweight
+  kernel gives features greater weight for distances closer to the point then the
+  Epanechnikov kernel does. Consequently, triweight results in “sharper” hotspots
+  and Epanechnikov results in “smoother” hotspots.
+
+  There are many shapes available (plese see the `Wikipedia page <https://en.wikipedia.org/wiki/Kernel_(statistics)#Kernel_functions_in_common_use>`_
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/interpolation.rst]: please

**Comment by ghtmtt** [comment]
@DelazJ I'm going to merge also this chapter if you agree

File: source/docs/user_manual/processing_algs/qgis/interpolation.rst (hunk 8 above)
[the remark this answers is not in the document]
Comment (DelazJ) [reply source/docs/user_manual/processing_algs/qgis/interpolation.rst]: Could you please create an issue report so that it's not forgotten?

**Pull request merged**